modules/swagger-codegen/src/main/resources/pkmst/cucumber/cucumberSteps.mustache

```diff
@@ -0,0 +1,76 @@
+package {{basePackage}}.cucumber;
+import cucumber.api.java.After;
+import cucumber.api.java.Before;
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+import cucumber.api.PendingException;
+
+public class {{serviceName}}Steps {
+
+@Given("^I want to write a step with precondition$")
+public void i_want_to_write_a_step_with_precondition() throws Throwable {
```

Comment:
What is the point of this ?

Reply:
Pkmst microservice  project will
a)Have all dependencies for writing BDD test using cucumber in POM
b)we will generate a sample .feature file (BDD driven test)
b)We will also generate a sample {servicename}steps file  ,making it easy for developer to go ahead and extend woth actual test code.

Following tests are autogenerated by pkmst.
a)Unit test via junit
b)Behaviour driven Test via cucumber
c)Performance Test autogenerated  via gatling

Reply:
pkmst microservice project provides support for cucumber based BDD. 
a)pkmst will generate a sample feature file  called Pkmst.feature
b)generate all the needed dependency in POM
c)generate a step file (Actual implementation of the feature file) file .
The step file  and  feature file is a place holder which development can go and edit as needed

For now we are making sure BDD support via cucmber with needed files and configurations exist.but developer needs to edit it for exct behaviour.

Next steps : we are looking at generating the .feature file and  step.java based on swagger. //in progress.